### PR TITLE
Plugin docs

### PR DESCRIPTION
**Pull request by hhamana**

### CODE REVIEW

File: docs/source/plugins.rst [hunk 1]
```diff
@@ -31,40 +45,148 @@ You add as many plugins as you want to your middleware. You pass them to the mid
...
+Do note headers are case-insentive.
+You can access the header value through the `<plugin class>.key` attribute,
+or through the `starlette_context.header_keys.HeaderKeys` enum.
 
-***********
-Date Header
-***********
 
-Extracts header "Date" and keeps it in context as a datetime.
++-----------------+----------------------+-------------------+-----------------------------------+
```
Comment (tomwojcik) [inline docs/source/plugins.rst:62]: Great idea, thank you!

File: docs/source/middleware.rst [hunk 2]
```diff
@@ -20,36 +56,10 @@ It is entirely possible that ``ContextMiddleware`` will be removed in the future
 It is also possible that authors will make some changes to the ``BaseHTTPMiddleware`` to fix this issue.
 I'd advise to only use ``RawContextMiddleware``.
 
-.. warning::
```
Comment (tomwojcik) [inline docs/source/middleware.rst:23]: I think this warning is still correct, so should not be removed. We only handle library errors. On 500 headers are not set on the response (sadly).

https://github.com/tomwojcik/starlette-context/blob/4550a190ac3e3b3c4dbdf9ba8733c7f78be80003/tests/test_context/test_context_on_unhandled_exception.py#L38-L40

**Review by tomwojcik — CHANGES_REQUESTED** [review]
Once you handle those 2 minor nitpicks I'll accept it as it. Wonderful job.

File: docs/source/middleware.rst (hunk 2 above)
Comment (hhamana) [reply docs/source/middleware.rst:23]: True, although at this point any Exception that arises would likely come from a user-implemented plugin not catching it.

File: docs/source/middleware.rst [hunk 3]
```diff
@@ -2,6 +2,42 @@
...
+The middleware effectively creates the context for the request, so you must configure your app to use it.
+More usage detail along with code examples can be found in :doc:`/plugins`.
+
+
+***********************************
+Errors and Middlewares in Starlette
+***********************************
+
+There may be a validation error occuring while processing the request in the plugins, which requires sending an error response.
+Starlette however does not let middleware use the regular error handler (`more details <https://www.starlette.io/exceptions/#errors-and-handled-exceptions>`_),
```
[the remark this answers is not in the document]
Comment (tomwojcik) [reply docs/source/middleware.rst]: @hhamana it seems like it was a temporary bug in Starlette and it's fixed now. Can you change the URL back? 🙈 

https://github.com/encode/starlette/issues/1212

[the remark this answers is not in the document]
Comment (hhamana) [reply docs/source/middleware.rst]: That explains why it was was fine when I wrote it. That it didn't resolve anymore had me fairly confused, but heh 🤷

**Review by tomwojcik — APPROVED** [review]

**Comment by tomwojcik** [comment]
v0.3.3 with your changes released :)

**Pull request merged**